File: Assets/Scripts/view/ScrollRectUIView.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using XLua;
using System;
using UnityEngine.UI;
public class ScrollRectUIView : MonoBehaviour
{
      public GameObject listContent;
      public GameObject scrollRect;
      public string partName;
     [HideInInspector]
     public ScrollRectControl SRC;
    void Awake()
    {
        SRC = transform.gameObject.GetComponent<ScrollRectControl>();
    }
    void Start()
    {
        scrollRect.GetComponent<ScrollRect>().onValueChanged.AddListener((value) => { SRC.onRecordDrag(value.y); });
    }

    [LuaCallCSharp]
    public void setRecordItem(int index,GameObject go,string itemName,Vector2 itemSize,Vector3 itemPos)
    {
        go.transform.SetParent(listContent.GetComponent<Transform>());
        RectTransform rt = go.GetComponent<RectTransform>();
        rt.pivot = new Vector2(0.5f, 1f);
        rt.anchoredPosition3D = itemPos;
        rt.sizeDelta = itemSize;
        go.SetActive(true);
        go.name = itemName;
        AvatarButton ab = go.GetComponent<AvatarButton>();
        ab.onButtonLeftClick += chengeMesh;
        ab.onButtonRightClick += AvatarControl._instance.remove;
        ab.onButtonRightClick += SRC.deleteItem;
        Sprite sprit = AvatarControl._instance.loadSpriteFromAssetBundle(go.name);
        go.GetComponent<Image>().sprite = sprit;
        go.GetComponent<Toggle>().group = scrollRect.GetComponent<ToggleGroup>();
 
    }
    public void chengeMesh(string state,string a ,string b){
        AvatarControl._instance.tryChangePeople(a,b);
    }
    public void setContent(int width, int height)
    {
        RectTransform rtf = listContent.GetComponent<RectTransform>();
        rtf.sizeDelta = new Vector2(width, height);
        rtf.localPosition = Vector3.zero;
   }
    [LuaCallCSharp]
    public void disableItem(GameObject go)
    {
        go.SetActive(false);
    }

    [LuaCallCSharp]
    public void destroyItem(GameObject go)
    {
        Destroy(go);
    }
}

File: Assets/.lpproj/Apis/Unity/CS/ScrollRectUIView.lua
---@class CS.ScrollRectUIView : CS.UnityEngine.MonoBehaviour
CS.ScrollRectUIView = {}

---@field public CS.ScrollRectUIView.listContent : CS.UnityEngine.GameObject
CS.ScrollRectUIView.listContent = nil

---@field public CS.ScrollRectUIView.scrollRect : CS.UnityEngine.GameObject
CS.ScrollRectUIView.scrollRect = nil

---@field public CS.ScrollRectUIView.partName : CS.System.String
CS.ScrollRectUIView.partName = nil

---@field public CS.ScrollRectUIView.SRC : CS.ScrollRectControl
CS.ScrollRectUIView.SRC = nil

---@return CS.ScrollRectUIView
function CS.ScrollRectUIView()
end

---@param y : CS.System.Single
function CS.ScrollRectUIView:onDrag(y)
end

---@param index : CS.System.Int32
---@param go : CS.UnityEngine.GameObject
---@param itemName : CS.System.String
---@param itemSize : CS.UnityEngine.Vector2
---@param itemPos : CS.UnityEngine.Vector3
function CS.ScrollRectUIView:setRecordItem(index, go, itemName, itemSize, itemPos)
end

---@param state : CS.System.String
---@param a : CS.System.String
---@param b : CS.System.String
function CS.ScrollRectUIView:chengeMesh(state, a, b)
end

---@param width : CS.System.Int32
---@param height : CS.System.Int32
function CS.ScrollRectUIView:setContent(width, height)
end

---@param go : CS.UnityEngine.GameObject
function CS.ScrollRectUIView:disableItem(go)
end